openmc/deplete/abc.py

```diff
@@ -701,6 +712,33 @@ def solver(self, func):
 
         self._solver = func
 
+    @property
+    def reactivity_control(self):
+        return self._reactivity_control
+
+    @reactivity_control.setter
+    def reactivity_control(self, reactivity_control):
+        if isinstance(reactivity_control, CellReactivityController):
+            if not isinstance(reactivity_control.cell, Cell):
+                raise TypeError(
+                    "Reactivity control attribute must be Cell "
+                    "not {}".format(type(reactivity_control.cell)))
+        elif isinstance(reactivity_control, MaterialReactivityController):
+            if not isinstance(reactivity_control.material, Material):
+                raise TypeError(
+                    "Reactivity control attribute must be Material "
+                    "not {}".format(type(reactivity_control.material)))
```

Comment:
Same comment on ensuring that there is no way to construct a `MaterialReactivityController` such that `material` is not a `Material`.

This allows us to shift the responsibility to check the validity off the client (`Operator`) and onto the construction of the thing it's using.